src/app/records/PresentationNodeLeaf.tsx

```diff
@@ -13,17 +15,103 @@ export default function PresentationNodeLeaf({
   ownedItemHashes,
   completedRecordsHidden,
   redactedRecordsRevealed,
+  sortRecordProgression,
 }: {
   node: DimPresentationNodeLeaf;
   ownedItemHashes?: Set<number>;
   completedRecordsHidden: boolean;
   redactedRecordsRevealed: boolean;
+  sortRecordProgression: boolean;
 }) {
+  const newNode = { ...node };
+  if (sortRecordProgression) {
+    /**
+     * RECORDS
+     */
+    if (node.records) {
+      newNode.records = orderBy(
```

Comment:
Rather than sorting and reassigning, you could assign the lists to local variables and conditionally sort them.